packages/mobile/e2e/src/usecases/ExchangeCelo.js

```diff
@@ -53,30 +53,28 @@ export default ExchangeCelo = () => {
     await element(by.id('ConfirmExchange')).tap()
     // Enter PIN
     await enterPinUiIfNecessary()
+    // Get transaction time
+    let transactionTime = new Date()
     // Scroll to top of feed
     await element(by.id('ExchangeScrollView')).scroll(200, 'down')
     // Wait 10 seconds checking that error banner is not visible each second
     await waitForExpectNotVisible('errorBanner')
-    // Assert balance is updated
-    await waitFor(element(by.id('CeloBalance')))
-      .toHaveText(`${(+celoBalanceBefore.text + CELO_TO_BUY).toFixed(3)}`)
-      .withTimeout(10 * 1000)
-    // Assert buy amount is present - displays amount after fees
-    await expect(
-      element(
-        by.text(`${(CELO_TO_BUY - FEES).toFixed(3)}`).withAncestor(by.id('TransactionList'))
-      ).atIndex(0)
-    ).toBeVisible()
+    // Wait up to 45 second and assert the transaction correctly appears
+    await waitFor(element(by.text(`${(CELO_TO_BUY - FEES).toFixed(3)}`)).atIndex(0))
```

Comment:
Is this FEES thing exact? If the fee varies a bit based on usage won't this fail?

Reply:
This will fail if fee varies. We could make a request to an external API to get the exact fees.

Reply:
I fear this adds one more thing that might fail and add flakiness 🤔  can we just trust that if _something_ shows up it has the correct value? Maybe send a random comment in the transaction and then check that there's a transaction with that comment?

Reply:
@gnardini I've added a check for the time string. Unfortunately, unlike send, exchanges don't have a message attached so adding a random string won't work in this case.

Reply:
Was able to use random amounts on sell and withdraw test; however, I had difficulties with the variable fees on buy. I've added a todo in the code and removed asserts on the amount. I've added a TODO to come back to this later.